src/hostcalls.rs

```diff
@@ -678,6 +678,66 @@ pub fn done() -> Result<(), Status> {
     }
 }
 
+extern "C" {
+    fn proxy_define_metric(
+        metric_type: MetricType,
+        name_data: *const u8,
+        name_size: usize,
+        return_id: *mut u32,
+    ) -> Status;
+}
+
+pub fn define_metric(metric_type: MetricType, name: &str) -> Result<u32, Status> {
+    let mut return_id: u32 = 0;
+    unsafe {
+        match proxy_define_metric(metric_type, name.as_ptr(), name.len(), &mut return_id) {
+            Status::Ok => Ok(return_id),
+            status => panic!("unexpected status: {}", status as u32),
+        }
+    }
+}
+
+extern "C" {
+    fn proxy_get_metric(metric_id: u32, return_value: *mut i64) -> Status;
```

Comment:
This operation is defined in `proxy-wasm-cpp-host` using `u64` (https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/e73c498e64c654c06a68ea54243e0cc9e2299464/include/proxy-wasm/exports.h#L84)

Why `i64` here ?

Reply:
Mistake. Thanks for catching it!